docs/book/src/topics/workload-identity.md

```diff
@@ -0,0 +1,204 @@
+      containerPath: /etc/kubernetes/pki/sa.pub
+    - hostPath: ${SERVICE_ACCOUNT_SIGNING_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.key
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+        service-account-issuer: ${SERVICE_ACCOUNT_ISSUER}
+        service-account-key-file: /etc/kubernetes/pki/sa.pub
+        service-account-signing-key-file: /etc/kubernetes/pki/sa.key
+    controllerManager:
+      extraArgs:
+        service-account-private-key-file: /etc/kubernetes/pki/sa.key
+EOF
+```
+
+- Initialize a management cluster using `clusterctl` using the below command.
+  If you do not have `clusterctl` installed, then follow this [link](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl)
+  to install.
+```bash
+$ clusterctl init --infrastructure azure
+```
+
+### Creating a Workload Cluster
+
+- Create a user-assigned identity using the below steps:
+  - [Create](https://learn.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azp#create-a-user-assigned-managed-identity)
+a user-assigned managed identity in Azure.
+  - [Create a role assignment](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal-managed-identity)
+to give the identity Contributor access to the Azure subscription where the workload cluster will be created.
+
+- Before generating a workload cluster YAML configuration set the
+  following environment variables.
+```bash
+export AZURE_SUBSCRIPTION_ID=<your-azure-subscription-id>
+# This is the client ID of the AAD app or user-assigned identity that you used to created the federated identity.
+export AZURE_CLIENT_ID=<your-azure-client-id>
+export AZURE_TENANT_ID=<your-azure-tenant-id>
+export AZURE_CONTROL_PLANE_MACHINE_TYPE="Standard_B2s"
+export AZURE_NODE_MACHINE_TYPE="Standard_B2s"
+export AZURE_LOCATION="eastus"
+
+# Identity secret. Though these are not used in workload identity, we still
+# need to set them for the sake of generating the workload cluster YAML configuration
+export AZURE_CLUSTER_IDENTITY_SECRET_NAME="cluster-identity-secret"
+export CLUSTER_IDENTITY_NAME="cluster-identity"
+export AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE="default"
+```
+- Generate a workload cluster template using the following command.
+
+```bash
+$ clusterctl generate cluster azwi-quickstart --kubernetes-version v1.27.3  --worker-machine-count=3 > azwi-quickstart.yaml
+```
+
+- Edit the generated `azwi-quickstart.yaml` to make the following changes for
+  workload identity to the `AzureClusterIdentity` object.
+  - Change the type to `WorkloadIdentity`.
+  - Remove the `clientSecret` spec.
+
+The AzureClusterIdentity specification should look like the following.
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureClusterIdentity
+metadata:
+  name: cluster-identity
```

Comment:
We also see in the `metadata`. Shall we mention to delete it?
```
metadata:
    name: cluster-identity
    namespace: default
```

Reply:
Did not understand?

Reply:
Apologies for the blunder. Corrected my question.

Reply:
That should be alright. It works either way and it is quite usual to explicitly specify default namespace.